interfaces/seccomp/backend.go

```diff
@@ -79,63 +78,12 @@ type Backend struct {
 	versionInfo seccomp.VersionInfo
 }
 
-// TODO: now that snap-seccomp has full support for deny-listing this
-// should be replaced with something like:
-//
-//	~ioctl - 4294967295|TIOCSTI
-//	~ioctl - 4294967295|TIOCLINUX
-//
-// in the default template. This requires that MaskedEq learns
-// to deal with two arguments (see also https://github.com/snapcore/snapd/compare/master...mvo5:rework-seccomp-denylist-incoperate-global.bin?expand=1)
-//
-// globalProfileLE is generated via cmd/snap-seccomp-black list
-var globalProfileLE = []byte{
-	0x20, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x15, 0x00, 0x00, 0x04, 0x3e, 0x00, 0x00, 0xc0,
-	0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x35, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x40,
-	0x15, 0x00, 0x00, 0x0d, 0xff, 0xff, 0xff, 0xff, 0x15, 0x00, 0x06, 0x0c, 0x10, 0x00, 0x00, 0x00,
-	0x15, 0x00, 0x00, 0x02, 0xb7, 0x00, 0x00, 0xc0, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x15, 0x00, 0x03, 0x09, 0x1d, 0x00, 0x00, 0x00, 0x15, 0x00, 0x00, 0x08, 0x15, 0x00, 0x00, 0xc0,
-	0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x15, 0x00, 0x00, 0x06, 0x36, 0x00, 0x00, 0x00,
-	0x20, 0x00, 0x00, 0x00, 0x1c, 0x00, 0x00, 0x00, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x15, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x00, 0x20, 0x00, 0x00, 0x00, 0x18, 0x00, 0x00, 0x00,
-	0x15, 0x00, 0x00, 0x01, 0x12, 0x54, 0x00, 0x00, 0x06, 0x00, 0x00, 0x00, 0x01, 0x00, 0x05, 0x00,
-	0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0x7f,
-}
-
-// globalProfileBE is generated via cmd/snap-seccomp-black list
-var globalProfileBE = []byte{
-	0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, 0x00, 0x15, 0x00, 0x08, 0x80, 0x00, 0x00, 0x16,
-	0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x15, 0x00, 0x06, 0x00, 0x00, 0x00, 0x36,
-	0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x18, 0x00, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-	0x00, 0x15, 0x00, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1c,
-	0x00, 0x15, 0x00, 0x01, 0x00, 0x00, 0x54, 0x12, 0x00, 0x06, 0x00, 0x00, 0x00, 0x05, 0x00, 0x01,
-	0x00, 0x06, 0x00, 0x00, 0x7f, 0xff, 0x00, 0x00,
-}
-
 // Initialize ensures that the global profile is on disk and interrogates
 // libseccomp wrapper to generate a version string that will be used to
 // determine if we need to recompile seccomp policy due to system
 // changes outside of snapd.
 func (b *Backend) Initialize(*interfaces.SecurityBackendOptions) error {
-	dir := dirs.SnapSeccompDir
-	fname := "global.bin"
-	var globalProfile []byte
-	if arch.Endian() == binary.BigEndian {
-		globalProfile = globalProfileBE
-	} else {
-		globalProfile = globalProfileLE
-	}
-	content := map[string]osutil.FileState{
-		fname: &osutil.MemoryFileState{Content: globalProfile, Mode: 0644},
-	}
-	if err := os.MkdirAll(dir, 0755); err != nil {
-		return fmt.Errorf("cannot create directory for seccomp profiles %q: %s", dir, err)
-	}
-	_, _, err := osutil.EnsureDirState(dir, fname, content)
```

Comment:
FWIW it's be nice to clean up the file, but there's no guarantee we're not racing with snap apps/services that could be starting up.  Maybe we need a follwup with some sort of ensure handler that effectively removes this file.

Reply:
Yes I thought about removing the file but I think it's really tricky and probably not worth it. I wonder how to track it so that we don't just forget?